### PR TITLE
Allow parallel iterators with param formals

### PR DESCRIPTION
**Pull request by vasslitvinov**
This newly-added test, mostly written by Engin, now works:

  test/functions/iterators/engin/paramArgIterator.chpl

DETAILS

preFold() converts a PRIM_TO_LEADER, PRIM_TO_FOLLOWER, or PRIM_TO_STANDALONE
to a CallExpr that resolves to a call to the corresponding parallel iterator.
The CallExpr's actuals are formals of the corresponding serial iterator.

When the user-written serial iterator has param formals, preFold() uses its
resolved+instantiated version. In that version, param formals are replaced
with default-intent formals with FLAG_INSTANTIATED_PARAM.

Prior to this change, those formals ended up as actuals of the CallExpr.
Since they are no longer params, the CallExpr did not resolve against
the parallel iterator with param formals.

To fix that, preFold() now looks them up in paramMap.

As the test shows, type formals work already.

### CODE REVIEW

**Comment by vasslitvinov** [comment]
@daviditen or @mppf would you review? It's quite simple.

**Comment by daviditen** [comment]
I can review.

File: test/functions/iterators/engin/paramArgIterator.good [hunk 1]
```diff
@@ -0,0 +1,8 @@
+par: 0
```
Comment (daviditen) [inline test/functions/iterators/engin/paramArgIterator.good]: The output of this test always comes in the same order. ser:0,1,2,3, par:0,10,20,30.

Can you fix the order of this .good file and remove the .prediff?

**Comment by daviditen** [comment]
Looks good other than the little test cleanup.

**Pull request merged**